From 0a6273d1cb1f985351f3e752f29bd9289ca529a5 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 10:12:45 -0400
Subject: [fs] ext4: use EINVAL if not a regular file in ext4_collapse_range()

Message-id: <1414059176-6044-22-git-send-email-lczerner@redhat.com>
Patchwork-id: 98289
O-Subject: [RHEL 7.1 PATCH 21/32] ext4: use EINVAL if not a regular file in ext4_collapse_range()
Bugzilla: 1150171

From: Theodore Ts'o <tytso@mit.edu>

BZ 1150171
https://bugzilla.redhat.com/show_bug.cgi?id=1150171

BREW 8149772
https://brewweb.devel.redhat.com/taskinfo?taskID=8149772

Upstream commit 86f1ca3889142d5959362c5694db3f3dc26f377a

 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/extents.c b/fs/ext4/extents.c
index 7c39d89..7fafdff 100644
--- a/fs/ext4/extents.c
+++ b/fs/ext4/extents.c
@@ -5393,7 +5393,7 @@ int ext4_collapse_range(struct inode *inode, loff_t offset, loff_t len)
   return -EINVAL;
 
  if (!S_ISREG(inode->i_mode))
-  return -EOPNOTSUPP;
+  return -EINVAL;
 
  trace_ext4_collapse_range(inode, offset, len);
 
-- 
1.7.1